ckanext/cnra_schema/cnra_schema_utils.py

```diff
@@ -0,0 +1,53 @@
+import json
+import logging
+
+from dateutil.parser import parse
+from six import text_type
+
+log = logging.getLogger(__name__)
+
+
+def delete_existing_extra_from_package_dict(key, package_dict):
+    existing_extra = {}
+    extras_list = package_dict.get('extras', [])
+    for extra in extras_list:
+        if extra['key'] == key:
+            existing_extra = extra
+
+    if existing_extra:
+        package_dict['extras'].remove(existing_extra)
```

Comment:
It is a bad idea to change the input data of a function. Please copy package_dict['extras'] to separate object and do modification on them.

Reply:
Thanks for catching this! I realized I had a variable `extras_list` that copied package_dict['extras'] to it. I've updated my PR to reflect this.